examples/sheep_wolf.jl

```diff
@@ -0,0 +1,164 @@
+using Agents, DataFrames, Random, StatsPlots
+
+mutable struct Sheep <: AbstractAgent
+    id::Int
+    pos::Tuple{Int,Int}
+    energy::Float64
+    reproduction_prob::Float64
+    Δenergy::Float64
+end
+
+mutable struct Wolf <: AbstractAgent
+    id::Int
+    pos::Tuple{Int,Int}
+    energy::Float64
+    reproduction_prob::Float64
+    Δenergy::Float64
+end
+
+mutable struct Grass <: AbstractAgent
+    id::Int
+    pos::Tuple{Int,Int}
+    fully_grown::Bool
+    regrowth_time::Int
+    countdown::Int
+end
+
+# randomize with the constraint that types must be together
+function by_breed(model::ABM)
+    c = [Int[], Int[], Int[]] # first=wolf, second=sheep
+    for a in allagents(model)
+        j = a isa Wolf ? 1 : a isa Sheep ? 2 : 3
+        push!(c[j], a.id)
+    end
+    for i in c; shuffle!(i); end
+    shuffle!(c)
+    return vcat(c...)
+end
+
+function initialize_model(;n_sheep=100, n_wolves=50, dims=(20,20), regrowth_time=30,
+    Δenergy_sheep=4, Δenergy_wolf=20, sheep_reproduce=.04, wolf_reproduce=.05)
+    space = GridSpace(dims)
```

Comment:
`space = GridSpace(dims, moore = true)`

At the moment, your random walk is just up, down, left, right.